sympy/algebras/tests/test_quaternion.py

```diff
@@ -46,6 +47,18 @@ def test_quaternion_construction_norm():
     assert (q1 * q3).norm() == q1.norm()
 
 
+def test_issue_25254():
+    # calculating the inverse cached the norm which caused problems
+    # when multiplying
+    p = Quaternion(1, 0, 1, 0)
+    q = Quaternion.from_axis_angle((1,1,1), 3 * math.pi/4)
+    qi = q.inverse()  # this operation cached the norm
+    try:
+        test = q * p * qi
```

Comment:
You can't assign the unused variable here.

Reply:
I simplified the test with something equivalent that would also fail without this change for the same reasons.

Reply:
Almost everything seems to be passing now, except for the tests related to Python 3.12. Is this enough?